pkg/auth/authz_handler.go

```diff
@@ -46,20 +49,123 @@ func (a *AuthzHandler) CreateAuthorizer() func(*http.Request) error {
 	return a.Authorizer
 }
 
-func (a *AuthzHandler) isObjectOwnedByUser(id string, obj interface{}, username string) (bool, error) {
+func (a *AuthzHandler) isTenancyEnabled() bool {
+	return a.OrgTenancyEnabled
+}
+
+func handleOwnershipQueryError(err error) (bool, error) {
+	if err != nil {
+		//if user is not the owner of the object return false
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return false, nil
+		}
+		//in case of a real db error, indicate it to the caller
+		return false, err
+	}
+	return true, nil
+}
+
+func (a *AuthzHandler) isObjectOwnedByUser(id string, obj interface{}, payload *ocm.AuthPayload) (bool, error) {
+	if a.db != nil {
+		err := a.db.First(obj, "id = ? and user_name = ?", id, payload.Username).Error
+		return handleOwnershipQueryError(err)
+	}
+
+	return true, nil
+}
+
+func (a *AuthzHandler) isObjectOwnedByOrg(id string, obj interface{}, payload *ocm.AuthPayload) (bool, error) {
 	if a.db != nil {
-		err := a.db.First(obj, "id = ? and user_name = ?", id, username).Error
+		err := a.db.First(obj, "id = ? and org_id = ?", id, payload.Organization).Error
+		return handleOwnershipQueryError(err)
+	}
+
+	return true, nil
+}
+
+func (a *AuthzHandler) hasOwnerAccess(id string, obj interface{}, payload *ocm.AuthPayload) (bool, error) {
+	if a.isTenancyEnabled() {
```

Comment:
perhaps here the best way is to ask first if object owned by user and only then own by org for backward comparability reasons

Reply:
yeah, could be cleaner as tenancy is a specific use-case.